### DISCUSSION TITLE
keep STAC related packages up to date

### DISCUSSION THREAD

**Discussion by joemcglinchy** [giswqs/geospatial#39] [Q&A] [answered]
Hi, not sure what all is needed to keep STAC related packages up to date, but following the installation recipe I am left with `pystac-client 0.3.0`. That packages is on version 0.7.5 right currently, but trying to install an updated version via mamba results in a whole mess of incompatibilities. using a `conda install pystac-client=0.7.5` results in a _very_ long solve, and `geospatial` being downgraded:

`geospatial                             0.8.0-pyhd8ed1ab_1 --> 0.2.2-pyhd8ed1ab_0`

**Comment by giswqs** [comment]
The recommended way is to use mamba rather than conda. The version restriction is likely caused by other dependencies. I need to look into it

**Comment by giswqs** [comment, marked as answer]
I just tried it on my Linux machine and found that it could install pystac-client v0.7.5. Please make sure you use mamba rather than conda. Conda takes forever to reslove a large mount of dependencies. Mamba is much faster. 

```bash
conda create -n geo python=3.11
conda activate geo
conda install -c conda-forge mamba
mamba install -c conda-forge geospatial
```

![image](https://github.com/opengeos/geospatial/assets/5016453/7fe05408-e362-4d48-8732-97252d4ec9bd)

**Comment by joemcglinchy** [comment]
yes, this looks good! thank you for the quick replies. I had to go `conda` earlier to keep with the python 3.9 instructions. I tried the same on my windows 10 machine:

```
conda create -n geo python=3.11
conda activate geo
conda install -c conda-forge mamba
mamba install geopandas
mamba install -c conda-forge geospatial
```
very fast. 

conda list | grep stac gives 👍 
```
pystac                    1.9.0              pyhd8ed1ab_0    conda-forge
pystac-client             0.7.5              pyhd8ed1ab_0    conda-forge
stack_data                0.6.2              pyhd8ed1ab_0    conda-forge
```

thank you!!
